### ISSUE TITLE
Reestablishing connection to the Lutron Bridge

### ISSUE THREAD

**Issue by MoTechnicalities** [thenewwazoo/homebridge-lutron-caseta-leap#82]
**Describe Your Problem:**
After moving my bridge around to find the best signal location, I noticed that when I reconnected the power/ethernet cord, Homebridge does not reconnect to the Lutron bridge to allow the Pico remotes to work with Homekit again until Homebridge gets manually restarted. Before I go any more in-depth, I'd just like to ask if this is the normal expectation. I do have a Reserved IP address for my Lutron Bridge. Can the plugin be made to look for and auto-reconnect to the bridge? Granted, it may not be normal for the bridge to disconnect. But if it does and then reconnects it would make for a little more fluent experience.

Other than that, GREAT plugin!

**Logs:**

```
Show the Homebridge logs here, remove any sensitive information.
```

**Plugin Config:**

```json
Show your Homebridge config.json here, remove any sensitive information.
```

**Screenshots:**

**Environment:**

* **Plugin Version**:
* **Homebridge Version**: 
* **Node.js Version**: 
* **NPM Version**: 
* **Operating System**:

**Comment by thenewwazoo** [comment]
My initial expectation is that it would reconnect when the ping interval goes, but tbh I don't actually know. Can you grab a log of an action, a disconnect, a reconnect, and then waiting for a few minutes? This might provide some insight into #31

**Comment by MoTechnicalities** [comment]
OK. I masked out info privy to my specific device and network, but here it is.
So, my log begins at 14:07:02. I used a Pico to control a light using a double press on button 1. Worked GREAT!
Then:
Between 14:07:29 & 14:08:39 I unplugged the power & Ethernet from the bridge momentarily. my Ring cam got me doing this.
The rest of the log is there too. but even after taking some time to edit/mask privy info and post. I have not rebooted Homebridge and do not have Pico functionality.
Hope it helps!


---MY LOGS----
[04/11/2022, 14:06:10] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:06:24] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:07:02] [Lutron] Button /button/101 on Pico remote Master Bedroom Pico got action Press
[04/11/2022, 14:07:02] [Lutron] btrk /button/101 got event Press in state 0
[04/11/2022, 14:07:02] [Lutron] btrk /button/101 now in state DOWN
[04/11/2022, 14:07:02] [Lutron] Button /button/101 on Pico remote Master Bedroom Pico got action Release
[04/11/2022, 14:07:02] [Lutron] btrk /button/101 got event Release in state 1
[04/11/2022, 14:07:02] [Lutron] btrk /button/101 cleared timer
[04/11/2022, 14:07:02] [Lutron] btrk /button/101 now in UP state
[04/11/2022, 14:07:03] [Lutron] Button /button/101 on Pico remote Master Bedroom Pico got action Press
[04/11/2022, 14:07:03] [Lutron] btrk /button/101 got event Press in state 2
[04/11/2022, 14:07:03] [Lutron] btrk /button/101 pressed before double-tap expiry
[04/11/2022, 14:07:03] [Lutron] btrk reset to IDLE
[04/11/2022, 14:07:03] [Lutron] button /button/101 got a double press
[04/11/2022, 14:07:03] [Lutron] Button /button/101 on Pico remote Master Bedroom Pico got action Release
[04/11/2022, 14:07:03] [Lutron] btrk /button/101 got event Release in state 0
[04/11/2022, 14:07:03] [Lutron] btrk /button/101 no-op IDLE action Release
[04/11/2022, 14:07:29] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:08:39] [Lutron] seeking bridge 02c0xxxx
[04/11/2022, 14:08:39] [Lutron] Bridge 02C0xxxx already known, closing.
[04/11/2022, 14:09:00] [Lutron] seeking bridge 02c0xxxx
[04/11/2022, 14:09:00] [Lutron] Bridge 02C0xxxx already known, closing.
[04/11/2022, 14:09:09] [Lutron] seeking bridge 02c0xxxx
[04/11/2022, 14:09:09] [Lutron] Bridge 02C0xxxx already known, closing.
[04/11/2022, 14:09:12] [Lutron] seeking bridge 02c0xxxx
[04/11/2022, 14:09:12] [Lutron] Bridge 02C0xxxx already known, closing.
[04/11/2022, 14:09:38] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:09:40] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:09:41] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [getPilot] Sending getPilot to a8bb50xxxxxx
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-68,"src":"","state":true,"sceneId":15,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-48,"src":"","state":true,"sceneId":11,"temp":2700,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-77,"src":"","state":true,"sceneId":15,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-70,"src":"","state":true,"sceneId":0,"r":0,"g":0,"b":255,"c":0,"w":0,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-54,"src":"","state":true,"sceneId":0,"temp":2700,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-49,"src":"","state":false,"sceneId":0,"r":0,"g":0,"b":255,"c":0,"w":0,"dimming":55}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-77,"src":"","state":true,"sceneId":11,"temp":2700,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-68,"src":"","state":false,"sceneId":11,"temp":2700,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-60,"src":"","state":false,"sceneId":6,"dimming":23}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-67,"src":"","state":false,"sceneId":19,"dimming":100}}
[04/11/2022, 14:10:23] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"getPilot","env":"pro","result":{"mac":"a8bb50xxxxxx","rssi":-65,"src":"","state":false,"sceneId":19,"dimming":100}}
[04/11/2022, 14:10:56] [Lutron] seeking bridge 02c0xxxx
[04/11/2022, 14:10:56] [Lutron] Bridge 02C0xxxx already known, closing.
[04/11/2022, 14:11:03] [Lutron] seeking bridge 02c0xxxx
[04/11/2022, 14:11:03] [Lutron] Bridge 02C0xxxx already known, closing.
[04/11/2022, 14:11:06] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:11:11] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:12:05] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:13:43] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:14:08] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:15:04] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:15:15] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:15:56] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:17:28] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:17:36] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:17:36] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:19:18] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:20:40] [Wiz] [SetPilot][192.168.xx.xx:38899] {"method":"setPilot","env":"pro","params":{"mac":"a8bb50xxxxxx","src":"","dimming":100,"rssi":-77,"state":false,"temp":2700}}
[04/11/2022, 14:20:40] [Wiz] [Socket] [192.168.xx.xx:38899] Received message: {"method":"setPilot","env":"pro","result":{"success":true}}
[04/11/2022, 14:21:14] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:24:52] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:25:42] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:25:45] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
connection failed:  Error: read ETIMEDOUT
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -110,
  code: 'ETIMEDOUT',
  syscall: 'read'
}
socket error: Error: read ETIMEDOUT
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -110,
  code: 'ETIMEDOUT',
  syscall: 'read'
}
client socket has closed.
[04/11/2022, 14:28:13] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:28:14] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:29:19] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:29:51] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:30:37] [Lutron] bridge 02c0xxxx got unsolicited message Response {
  Header: {
    StatusCode: ResponseStatus { message: 'NoContent', code: 204 },
    Url: '/device/status/deviceheard',
    MessageBodyType: undefined
  },
  CommuniqueType: 'SubscribeResponse',
  Body: undefined
}
[04/11/2022, 14:30:37] [Lutron] bridge 02c0xxxx got unsolicited message Response {
  Header: {
    StatusCode: ResponseStatus { message: 'NoContent', code: 204 },
    Url: '/zone/status/deprecated/level',
    MessageBodyType: undefined
  },
  CommuniqueType: 'SubscribeResponse',
  Body: undefined
}
[04/11/2022, 14:30:46] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:32:26] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:34:36] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:34:40] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:36:02] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:37:49] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:37:54] [Ring] Kitchen Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:38:05] [Ring] Upstairs Detected Motion. Loading snapshot before sending event to HomeKit
[04/11/2022, 14:38:56] [Ring] Family Room Detected Motion. Loading snapshot before sending event to HomeKit

**Comment by MoTechnicalities** [comment]
A reboot after the above post has put things back in action.

**Comment by thenewwazoo** [comment]
Okay, I'm pretty sure this is fixed in 2.4.5-beta0.

**Comment by MoTechnicalities** [comment]
It's a beautiful thing you've done here! Now, After unplugging and reconnecting the bridge it always returns to rock-solid and as-expected performance.

I use my Picos to issue HomeKit SSH shortcut commands to my WiZ lights. The WiZ plugin had its own problems too with the bulbs falling asleep on the network. But now that that is also fixed... Man! This plugin is so sweet!🎉🎉🎉

**Lutron should pay you for the extra Picos I'm going to purchase!**

**Comment by stevenmadow** [comment]
Hey - I am experiencing this same issue on 2.4.5 - beta 1.  Any ideas @thenewwazoo ?

**Comment by thenewwazoo** [comment]
Should be fixed in #119!
